opal/mca/accelerator/base/accelerator_base_select.c

```diff
@@ -32,116 +32,108 @@
-        if (NULL == aoc) {
-            return OPAL_ERR_OUT_OF_RESOURCE;
-        }
-        OBJ_CONSTRUCT(aoc, opal_list_item_t);
-        aoc->accelerator_component = component;
-        aoc->accelerator_module = module;
-        opal_list_append(&opened, (opal_list_item_t*) aoc);
-    }
-
-    /* Finished querying all components. Check for no components found */
-
-    if( NULL == best_component ) {
-        return OPAL_ERR_NOT_FOUND;
-    }
-
-    opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                         "selected %s best priority %d\n",
-                         best_component->base_version.mca_component_name, best_priority);
-
-    /* Save the winner */
-    accelerator_base_selected_component = *best_component;
-    opal_accelerator = *best_module;
-
-    /* Finalize all non-selected components */
 
-    for (item = opal_list_remove_first(&opened);
-         NULL != item;
-         item = opal_list_remove_first(&opened)) {
-        aoc = (accelerator_opened_component_t *) item;
-
-        if (aoc->accelerator_component != best_component ) {
-            /* Finalize */
-
-            if (NULL != aoc->accelerator_component->accelerator_finalize) {
-
-                /* Blatently ignore the return code (what would we do to
-                   recover, anyway?  This component is going away, so errors
-                   don't matter anymore) */
-
-                aoc->accelerator_component->accelerator_finalize(aoc->accelerator_module);
-                opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
-                                    "select: component %s not selected / finalized",
-                                    aoc->accelerator_component->base_version.mca_component_name);
+        module = component->accelerator_init();
+        if (NULL != module) {
+            /* A second non NULL component (the final entry in the list) has initialized successfully */
+            if (found_component) {
+                if (0 != ordered_list.opal_list_length) {
+                    opal_show_help("help-accelerator-base.txt", "Multiple Accelerators Found", true,
+                                   accelerator_base_selected_component.base_version.mca_component_name,
+                                   component->base_version.mca_component_name);
+
+                    /* Don't leak the two initalized components */
+                    accelerator_base_selected_component.accelerator_finalize(&opal_accelerator);
+                    component->accelerator_finalize(module);
+
+                    /* Throw a fatal error, can not go on without knowing which Accelerator to use
+                    Leak some memory but doesn't matter b/c OMPI is aborting */
+                    return OPAL_ERR_FATAL;
+                }
+
+                /* Prefer found component over NULL component*/
+                // OBJ_DESTRUCT(item);
```

Comment:
Leftover code?

Reply:
no, I actually want this.  Commented it out when I was debugging a seg fault, but forgot to put this back in.